pkg/agent/openflow/pipeline.go

```diff
@@ -847,6 +851,206 @@ func (f *featureService) snatConntrackFlows() []binding.Flow {
 	return flows
 }
 
+func matchTransportHeader(packet *binding.Packet, flowBuilder binding.FlowBuilder, endpointPackets []binding.Packet) binding.FlowBuilder {
+	// Match transport header
+	switch packet.IPProto {
+	case protocol.Type_ICMP:
+		flowBuilder = flowBuilder.MatchProtocol(binding.ProtocolICMP)
+	case protocol.Type_IPv6ICMP:
+		flowBuilder = flowBuilder.MatchProtocol(binding.ProtocolICMPv6)
+	case protocol.Type_TCP:
+		if packet.IsIPv6 {
+			flowBuilder = flowBuilder.MatchProtocol(binding.ProtocolTCPv6)
+		} else {
+			flowBuilder = flowBuilder.MatchProtocol(binding.ProtocolTCP)
+		}
+	case protocol.Type_UDP:
+		if packet.IsIPv6 {
+			flowBuilder = flowBuilder.MatchProtocol(binding.ProtocolUDPv6)
+		} else {
+			flowBuilder = flowBuilder.MatchProtocol(binding.ProtocolUDP)
+		}
+	default:
+		flowBuilder = flowBuilder.MatchIPProtocolValue(packet.IsIPv6, packet.IPProto)
+	}
+	if packet.IPProto == protocol.Type_TCP || packet.IPProto == protocol.Type_UDP {
+		if endpointPackets != nil && endpointPackets[0].DestinationPort != 0 {
+			flowBuilder = flowBuilder.MatchDstPort(endpointPackets[0].DestinationPort, nil)
+		} else if packet.DestinationPort != 0 {
+			flowBuilder = flowBuilder.MatchDstPort(packet.DestinationPort, nil)
+		}
+		if packet.SourcePort != 0 {
+			flowBuilder = flowBuilder.MatchSrcPort(packet.SourcePort, nil)
+		}
+	}
+
+	return flowBuilder
+}
+
+// flowsToCapture generates flows for packet capture. dataplaneTag is used as a mark for the target flow.
+func (f *featurePodConnectivity) flowsToCapture(dataplaneTag uint8,
+	ovsMetersAreSupported,
+	senderOnly bool,
+	receiverOnly bool,
+	packet *binding.Packet,
+	endpointPackets []binding.Packet,
+	ofPort uint32,
+	timeout uint16) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(cookie.PacketCapture).Raw()
+	var flows []binding.Flow
+	tag := uint32(dataplaneTag)
+	var flowBuilder binding.FlowBuilder
+	if !receiverOnly {
+		// if not receiverOnly, ofPort is inPort
+		if endpointPackets == nil {
+			flowBuilder = ConntrackStateTable.ofTable.BuildFlow(priorityHigh).
+				Cookie(cookieID).
+				MatchInPort(ofPort).
+				MatchCTStateTrk(true).
+				Action().LoadToRegField(PacketCaptureMark, tag).
+				SetHardTimeout(timeout).
+				Action().GotoStage(stagePreRouting)
+			if packet.DestinationIP != nil {
+				flowBuilder = flowBuilder.MatchDstIP(packet.DestinationIP)
+			}
```

Comment:
It doesn't match any metadata other than destination IP, how could the `packet` filter work?

Reply:
I have create a function to match fields like port,protocol in as `matchTransportHeader`. As i have installed multiple flows and this can be shared. the flowBuilder is not `Done` here.